internal/controllers/topology/cluster/patches/engine.go

```diff
@@ -302,29 +303,29 @@ func applyPatchesToRequest(ctx context.Context, req *runtimehooksv1.GeneratePatc
 			log.V(5).Infof("Accumulating JSON patch: %s", string(patch.Patch))
 			jsonPatch, err := jsonpatch.DecodePatch(patch.Patch)
 			if err != nil {
-				return errors.Wrapf(err, "failed to apply patch with uid %q: error decoding json patch (RFC6902): %s",
-					requestItem.UID, string(patch.Patch))
+				log.V(2).Infof("Failed to apply patch with uid %q: error decoding json patch (RFC6902): %s: %s", requestItem.UID, string(patch.Patch), err)
+				return errors.Wrap(err, "failed to apply patch: error decoding json patch (RFC6902)")
```

Comment:
Q: could we keep adding the patch to the error and drop only the uid?

also lets point users to the logs + add a note why are doing this
```suggestion
				// NOTE: details about the error are not included in the error because uid changes at every reconcile.
				log.V(2).Infof("Failed to apply patch with uid %q: error decoding json patch (RFC6902): %s: %s", requestItem.UID, string(patch.Patch), err)
				return errors.Wrap(err, "failed to apply patch: error decoding json patch (RFC6902); see Cluster API controller logs for more details")
```

Reply:
There's a conversation about it here: https://github.com/kubernetes-sigs/cluster-api/pull/7345#discussion_r987979604

It comes down to if the patch were to change every reconcile (possible but probably not the most common situation) then the same problem would emerge. Here the logging is on L2 so it seems clear enough which patch is at issue. 

Instead of pointing users to the logs here a "Look at the logs" message could be added to every error condition (maybe just for TopologyReconciled). IMO this isn't something specific to this part of the code.

Alternatively reconciliation could be filtered for when the condition changes, or rate limiting would massively reduce the impact here. It seems likely this issue will recur again in future as it's a very easy mistake to make during reconcile.

Reply:
> Instead of pointing users to the logs here a "Look at the logs" message could be added to every error condition (maybe just for TopologyReconciled). IMO this isn't something specific to this part of the code.

I think this could be possibly addressed by having a systematic troubleshooting guide like:
* use clusterctl describe
* identify the failed resource
* look at the conditions
* look at the corresponding log (if necessary)
* ...

Reply:
Let me open another issue for that and take it out of this discussion if that's good with everyone?

Reply:
That was my plan :)

Reply:
I don't want to block this, but I think that adding this message won't hurt when we are omitting intentionally details from the error. We can eventually iterate later on this one.

Reply:
I'd rather not add it here TBH as it's a very general idea - we should make it very clear elsewhere what the troubleshooitng process should be instead of trying to hind at it in error conditions.